insights/parsers/mongod_conf.py

```diff
@@ -0,0 +1,93 @@
+"""
+mongodb-server configuration files
+==================================
+
+This module contains the following files:
+    ``/etc/mongod.conf`` and ``/etc/mongodb.conf``.
+
+Both of them are provided by package mongodb-server.
+This MongoDB configuration file may use the **YAML** format
+or the standard **key-value pair** format.
+
+
+Sample input(YAML format)::
+
+    systemLog:
+      destination: file
+      logAppend: true
+      path: /var/log/mongodb/mongod.log
+
+    # Where and how to store data.
+    storage:
+      dbPath: /var/lib/mongo
+      journal:
+        enabled: true
+
+
+Sample input(key-value pair format)::
+
+    # mongodb.conf - generated from Puppet
+    #where to log
+    logpath=/var/log/mongodb/mongodb.log
+    logappend=true
+    # Set this option to configure the mongod or mongos process to bind to and
+    # listen for connections from applications on this address.
+    # You may concatenate a list of comma separated values to bind mongod to multiple IP addresses.
+    bind_ip = 127.0.0.1
+    # fork and run in background
+    fork=true
+    dbpath=/var/lib/mongodb
+    # location of pidfile
+    pidfilepath=/var/run/mongodb/mongodb.pid
+    # Enables journaling
+    journal = true
+    # Turn on/off security.  Off is currently the default
+    noauth=true
+
+
+Examples:
+
+    >>> mongod_conf = shared[MongodConf]
+    >>> 'storage' in mongod_conf.data
+    True
+    >>> mongod_conf.data['storage']['dbPath']
+    '/var/lib/mongo'
+    >>> mongod_conf.get("bind_ip")
+    '127.0.0.1'
+
+"""
+
+import yaml
+from .. import parser, Parser, LegacyItemAccess, get_active_lines
+from ..parsers import ParseException, split_kv_pairs
+
+
+@parser('mongod.conf')
+@parser('mongod.conf_rh_mongodb26')
+@parser('mongodb.conf')
+class MongodbConf(Parser, LegacyItemAccess):
+    """
+    Read the ``/etc/mongod.conf`` YAML file.
+
+    Raises:
+        ParseException: Raised when any problem parsing the file content.
+    """
+    def parse_content(self, content):
+        kv_type = self._file_type_is_kv(content)
+        try:
+            if kv_type:
+                self.data = split_kv_pairs(content, use_partition=True)
+            else:
+                self.data = yaml.safe_load('\n'.join(content))
+        except Exception, e:
+            raise ParseException('mongod conf parse faild: %s', e)
+
+    def _file_type_is_kv(self, content):
+        """
+        Return True if this file type is key-value. Return False means
+        this file will be handled in YAML format.
+        """
+        content = get_active_lines(content)
+        cnt = sum([1 for line in content if "=" in line])
+        percent = float(cnt) / len(content)
+        return True if percent > 0.9 else False
```

Comment:
Better add more explanation for this logic (e.g. why 90%?).

Reply:
Sure.